sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeAppendFileClient.cs

```diff
@@ -0,0 +1,752 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Storage.Files.DataLake.Models;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Files.DataLake
+{
+    /// <summary>
+    /// The <see cref="DataLakeFileClient"/> allows you to manipulate Azure Data Lake append blob-based files.
+    /// </summary>
+    public class DataLakeAppendFileClient : DataLakeFileClient
```

Comment:
I understand where this name is coming from but I think this might be confusing for customers without much context.
I.e. `DataLakeFileClient` exposes some `Append` APIs then we have `DataLakeAppendFileClient` which exposes same `Append` APIs and one extra. We're trying to tell that one is backed by different blob type than another, but then we call all data writes after `Append`.
On the other hand I'm having hard time suggesting better naming. Maybe `DataLakeConcurrentFileClient`/`DataLakeThreadSafeFileClient`/`DataLakeParallelAccessFileClient` ?
We may want to have more folks to chime in on naming this.

Reply:
I'm going to leave it for now, until we come up with a better name.